lib/lzstring.dart

```diff
@@ -50,200 +61,162 @@ class LZString {
+          String? input) async =>
+      decompressFromEncodedURIComponentSync(input);
+
+  /// Synchronously decompress ASCII strings [input] which produces by
+  /// [compressToEncodedURIComponent].
+  static String? decompressFromEncodedURIComponentSync(String? input) {
+    if (input?.isEmpty ?? true) return null;
+    input = input!.replaceAll(' ', '+');
+    return _decompress(input.length, 32,
         (index) => _getBaseValue(_keyStrUriSafe, input![index]));
   }
 
-  /// Produces ASCII strings representing the original string encoded in Base64 with a few tweaks to make these URI safe.
+  /// Produces ASCII strings representing the original string encoded in Base64
+  /// with a few tweaks to make these URI safe.
+  ///
+  /// Can be decompressed with [decompressFromEncodedURIComponent]
+  static Future<String?> compressToEncodedURIComponent(String? input) async =>
+      compressToEncodedURIComponentSync(input);
+
+  /// Synchronously produces ASCII strings representing the original string
+  /// encoded in Base64 with a few tweaks to make these URI safe.
   ///
   /// Can be decompressed with `decompressFromEncodedURIComponent`
-  static Future<String> compressToEncodedURIComponent(String? input) async {
-    if (input == null) "";
-    return await _compress(input, 6, (a) => _keyStrUriSafe[a]);
-  }
+  static String? compressToEncodedURIComponentSync(String? input) =>
+      _compress(input, 6, (a) => _keyStrUriSafe[a]);
 
   /// Produces invalid UTF-16 strings from [uncompressed].
   ///
   /// Can be decompressed with `decompress`.
   ///
-  static Future<String> compress(final String? uncompressed) async {
-    return await _compress(uncompressed, 16, (a) => String.fromCharCode(a));
-  }
+  static Future<String?> compress(final String? uncompressed) async =>
+      compressSync(uncompressed);
 
-  static Future<String> _compress(
-      String? uncompressed, int bitsPerChar, GetCharFromInt getCharFromInt) {
-    return Future<String>(() {
-      if (uncompressed == null) return "";
-      int? value;
-      Map<String, int> contextDictionary = Map<String, int>();
-      Map<String, bool> contextDictionaryToCreate = Map<String, bool>();
-      String contextC = "";
-      String contextWC = "";
-      String contextW = "";
-      int contextEnlargeIn =
-          2; // Compensate for the first entry which should not count
-      int contextDictSize = 3;
-      int contextNumBits = 2;
-      StringBuffer contextData = StringBuffer();
-      int contextDataVal = 0;
-      int contextDataPosition = 0;
-      int ii;
-
-      for (ii = 0; ii < uncompressed.length; ii++) {
-        contextC = uncompressed[ii];
-        if (!contextDictionary.containsKey(contextC)) {
-          contextDictionary[contextC] = contextDictSize++;
-          contextDictionaryToCreate[contextC] = true;
-        }
+  /// Synchronously produces invalid UTF-16 strings from [uncompressed].
+  ///
+  /// Can be decompressed with `decompress`.
```

Comment:
Could you also update all inline code block to square brackets in rest of the doc?

Reply:
Sure thing

Reply:
Fixed